env_functions.sh

```diff
@@ -30,7 +30,8 @@ hg_prompt() {
   repo=$(prompt::_find_repo .hg) || return 0
   cd "$repo" || return # so Mercurial doesn't have to do the same find we just did
   local branch num_heads heads
-  branch=$(hg branch 2> /dev/null) || return 0
+  # `hg branch` may be slow for large repos, read it from .hg
+  { branch=$(<.hg/branch) || printf default; } 2>/dev/null
```

Comment:
Why did you do this and not the syntax I suggested? If I'm reading this right it's going to `printf default` directly to stdout, not to the `branch` variable.

Reply:
Right, I wasn't able to get `branch=$({ < .hg/branch; } 2>/dev/null || echo default)` working locally and accidentally uploaded this version.

And doesn't seem that $(<file) can have || statements there?

Reply:
Ah `$(<file)` works but it looks like `$({<file})` or any other more complex pipeline outputs nothing. Sorry for the misdirection, I was just trying to avoid a [UUOC](https://en.wikipedia.org/wiki/Cat_(Unix)#Useless_use_of_cat), which is probably overkill in any case.